GamePortal/GamePortal.Web.Api/Controllers/Sapper/SapperGameController.cs

```
@@ -47,9 +45,23 @@ public IHttpActionResult UserInfoById(int id)
         /// </summary>
         /// <returns>User id</returns>
         [HttpPost, Route("users/add")]
-        public IHttpActionResult UserAdd(int id)
+        public IHttpActionResult UserAdd([FromBody] UserDto model)
         {
-            return Ok("User add: id = " + id);
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            try
+            {
+                var result = _sapperService.UserAdd(model);
+                return Created($"/users/{result.Id}", result.Id);
+            }
+            catch (Exception e)
```

don't catch all exceptions here